tests/Doc/QueryAggExample.cs

```diff
@@ -0,0 +1,304 @@
+        Assert.Equal(
+            "Key: bicycle:8, Price: 1200, Discounted: 1080",
+            $"Key: {test1Row["__key"]}, Price: {test1Row["price"]}, Discounted: {test1Row["discounted"]}"
+        );
+        // REMOVE_END
+
+
+        // STEP_START agg2
+        AggregationResult res2 = db.FT().Aggregate(
+            "idx:bicycle",
+            new AggregationRequest("*")
+                .Load(new FieldName("price"))
+                .Apply("@price<1000", "price_category")
+                .GroupBy(
+                    "@condition",
+                    Reducers.Sum("@price_category").As("num_affordable")
+                )
+        );
+        Console.WriteLine(res2.TotalResults);   // >>> 3
+
+        for (int i = 0; i < res2.TotalResults; i++)
+        {
+            Row res2Row = res2.GetRow(i);
+
+            Console.WriteLine(
+                $"Condition: {res2Row["condition"]}, Num affordable: {res2Row["num_affordable"]}"
+            );
+        }
+        // >>> Condition: refurbished, Num affordable: 1
+        // >>> Condition: used, Num affordable: 1
+        // >>> Condition: new, Num affordable: 3
+        // STEP_END
+
+        // Tests for 'agg2' step.
+        // REMOVE_START
+        Assert.Equal(3, res2.TotalResults);
+
+        Row test2Row = res2.GetRow(0);
+        Assert.Equal(
+            "Condition: refurbished, Num affordable: 1",
+            $"Condition: {test2Row["condition"]}, Num affordable: {test2Row["num_affordable"]}"
+        );
+
+        test2Row = res2.GetRow(1);
+        Assert.Equal(
+            "Condition: used, Num affordable: 1",
+            $"Condition: {test2Row["condition"]}, Num affordable: {test2Row["num_affordable"]}"
+        );
+
+        test2Row = res2.GetRow(2);
+        Assert.Equal(
+            "Condition: new, Num affordable: 3",
+            $"Condition: {test2Row["condition"]}, Num affordable: {test2Row["num_affordable"]}"
+        );
+        // REMOVE_END
+
+
+        // STEP_START agg3
+        AggregationResult res3 = db.FT().Aggregate(
+            "idx:bicycle",
+            new AggregationRequest("*")
+                .Apply("'bicycle'", "type")
+                .GroupBy("@type", Reducers.Count().As("num_total"))
+        );
+        Console.WriteLine(res3.TotalResults);   // >>> 1
+
+        Row res3Row = res3.GetRow(0);
+        var res = $"Type: {res3Row["type"]}, Num total: {res3Row["num_total"]}";
+        Console.WriteLine($"Type: {res3Row["type"]}, Num total: {res3Row["num_total"]}");
+        // >>> Type: bicycle, Num total: 10
+        // STEP_END
+
+        // Tests for 'agg3' step.
+        // REMOVE_START
+        Assert.Equal(1, res3.TotalResults);
+
+        Assert.Equal(
+            "Type: bicycle, Num total: 10",
+            $"Type: {res3Row["type"]}, Num total: {res3Row["num_total"]}"
+        );
+        // REMOVE_END
+
+
+        // STEP_START agg4
+
+        // Not supported in NRedisStack.
+
+        // STEP_END
```

Comment:
could you share the code fragment you tried?

Reply:
@atakavci I've added it after `// STEP_START agg4` as a separate commit. It basically just doesn't include the list of bicycles that were supposed to come from the ToList reducer. The Row object doesn't even seem to have an element for the bicycles list.

Reply:
@andy-stark-redis you are right ? we have a bug with getting list of items with result. 
To share more context, return type does not provide a way to contain a list of items as of now _(it is a Dictionary<string, RedisValue>)_. We might need to introduce a breaking change to fix this. 
Would you mind opening an issue for this? i ll pick it soon.

Reply:
@atakavci I've added [this issue](https://github.com/redis/NRedisStack/issues/342). I've replaced the dodgy code with the original missing example message. I'll edit the issue to add a link to the source file once this PR is merged.

Reply:
great! thanks..